app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@stanhu the test of file replacement can be achieved by using md5 check, like:
  step 'I see the new content of the ".gitignore"' do
    Digest::MD5.hexdigest("LICENSE") == Digest::MD5.hexdigest(".gitignore")
    Digest::MD5.hexdigest(".gitignore") != Digest::MD5.hexdigest(old_gitignore_content)
  end
Do you have any suggestions regarding to the uploaded file size ? Thanks.